icu4c/source/i18n/dtitvinf.cpp

```diff
@@ -426,8 +427,25 @@ DateIntervalInfo::initializeData(const Locale& locale, UErrorCode& status)
         calTypeBundle = ures_getByKeyWithFallback(calBundle, calendarTypeToUse, NULL, &status);
         itvDtPtnResource = ures_getByKeyWithFallback(calTypeBundle,
                                                      gIntervalDateTimePatternTag, NULL, &status);
+        // TODO(ICU-20400): After the fixing, we should find the "fallback" from
+        // the rb directly by the path "calendar/${calendar}/intervalFormats/fallback".
         resStr = ures_getStringByKeyWithFallback(itvDtPtnResource, gFallbackPatternTag,
                                                  &resStrLen, &status);
+        if ( U_FAILURE(status) ) {
```

Comment:
good point, so I will change the code. But I cannot leave the failure there otherwise, it won't do the following task. I also need to change the status to U_USING_FALLBACK_WARNING in the case it succeed.